services/gem/gem-helpers.js

```diff
@@ -0,0 +1,25 @@
+import { valid, maxSatisfying, prerelease } from '@renovatebot/ruby-semver'
+
+function latest(versions) {
+  // latest Ruby Gems version, including pre-releases
+  return maxSatisfying(versions, '>0')
+}
```

Comment:
I haven't implemented including/excluding pre-releases here - for stable version only we just report the `.version` param from the API so we only actually need the 'including pre-releases' case.

Reply:
Do you know offhand if this will support or typical behavior of falling back to a prerelease version in cases where the user has requested stable-only versions but all the versions are prerelease?

Reply:
As I say, I haven't implemented `includePre` here because we don't need to use this function if the user wants latest stable.
If we want the latest stable, we just return the value of `.version` here: https://github.com/badges/shields/blob/d8c3dd12e2eb8b1afd9543783fb6d894e198ed61/services/gem/gem-version.service.js#L71-L73
`maxSatisfying()` doesn't care about stable/pre-releases. It just returns the latest version that matches the constraint, so if we wanted to implement a version that returns the latest stable, we'd do it something like (not actually tested, but hopefully you get the point):

```js
function latest(versions, includePre) {
  const stableVersions = versions.filter((v) => !prerelease(v) && v[0] !== '0')
  if (stableVersions.length >0 && !includePre) {
   return maxSatisfying(stableVersions, '>0') 
  }
  return maxSatisfying(versions, '>0')
}
```

but we don't need it
so I haven't bothered

Reply:
got it. apologies it wasn't clear to me on first read what you meant in the initial comment but i see it now :+1:

Reply:
no probs